File: NOVEMBER-2025/29-11-2025/Count set bits.py
class Solution:
    def countSetBits(self, n):
        count = 0
        while n > 0:
            x = n.bit_length() - 1  # position of MSB
            if x == 0:
                count += 1
                break
            count += x * (1 << (x - 1))
            count += n - (1 << x) + 1
            n -= (1 << x)
        return count

File: NOVEMBER-2025/29-11-2025/Count set bits.js
class Solution {
    countSetBits(N) {
        let count = 0;
        while (N > 0) {
            let x = Math.floor(Math.log2(N)); // position of MSB
            count += x * (1 << (x - 1));
            count += N - (1 << x) + 1;
            N -= (1 << x);
        }
        return count;
    }
}